### ISSUE TITLE
Missing Player CompassTarget

### ISSUE THREAD

**Issue by ST-DDT** [SpongePowered/SpongeAPI#554]
In Bukkit there was a very nice feature to set the compass target for each player individually.
[See Bukkit's Player](https://github.com/Bukkit/Bukkit/blob/master/src/main/java/org/bukkit/entity/Player.java#L79-91)
I would like to see something similar for Sponge as well.

Here a list of things i thought of that could be useful:
- Get/Set/Reset the player's compass target 
  - What about persistence here?
  - Possible to set for remote worlds?
- Get/Set/Reset the world's compass target
- Automatic compass targeting strategies: SPAWN, BED...
  - Is this even possible?

When combined with the Data API PR this will result in only a single additional (API) class.

Comments? Suggestions?

**Comment by modwizcode** [comment]
I'm agreeing with this being tied into the Data API and being persistent. It should be possible to have built-in default targets but setting the compass manually to those targets is really not easier enough to justify it.

The world compass target needs more discussion, I have no issue with it, but it seems like that might make it much more confusing to work with.

Additionally it should probably have a setHeading(double rotation) that can set a constant heading direction on the compass so plugins can avoid the annoying location management that would be needed to do completely custom compass control.

**Comment by gabizou** [comment]
Theoretically, it can be handled by [`TargetedLocationData`](https://github.com/SpongePowered/SpongeAPI/pull/542/files#diff-45830d910c7b7ef337bcfc8f81439d9dR30). However, implicitly defining that in #542 is out of scope of that PR. I do believe that it should be possible in a new PR after #542 is merged.

**Comment by modwizcode** [comment]
@gabizou I was going to add a new label representing dependence on a major API change to flag this. I agree it should be done after #542

**Comment by Aaron1011** [comment]
This is provided by `TargetedLocationData`
